homeassistant/components/tado/weather.py

```diff
@@ -0,0 +1,110 @@
+"""Support for the Tado weather service."""
+from homeassistant.components.weather import WeatherEntity
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.const import TEMP_CELSIUS, TEMP_FAHRENHEIT
+from homeassistant.core import HomeAssistant, callback
+from homeassistant.helpers.dispatcher import async_dispatcher_connect
+
+from .const import (
+    ATTRIBUTION,
+    CONDITIONS_MAP,
+    DATA,
+    DOMAIN,
+    SIGNAL_TADO_UPDATE_RECEIVED,
+)
+from .entity import TadoHomeEntity
+
+
+async def async_setup_entry(
+    hass: HomeAssistant, entry: ConfigEntry, async_add_entities
+):
+    """Set up the Tado weather platform."""
+
+    tado = hass.data[DOMAIN][entry.entry_id][DATA]
+
+    async_add_entities([TadoWeatherEntity(tado, hass.config.units.is_metric)], True)
+
+
+def format_condition(condition: str) -> str:
+    """Return condition from dict CONDITIONS_MAP."""
+    for key, value in CONDITIONS_MAP.items():
+        if condition in value:
+            return key
+    return condition
+
+
+class TadoWeatherEntity(TadoHomeEntity, WeatherEntity):
+    """Define a Tado weather entity."""
+
+    def __init__(self, tado, is_metric):
+        """Initialize."""
+        self._tado = tado
+        self._data = self._tado.data["weather"]
+        super().__init__(tado)
```

Comment:
Shouldn't the super constructor be called first?

Reply:
Yes, should be fixed now.